collect_app/src/main/java/org/odk/collect/android/formentry/repeats/DeleteRepeatDialogFragment.java

```diff
@@ -0,0 +1,72 @@
+package org.odk.collect.android.formentry.repeats;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.javarosawrapper.FormController;
+
+import static android.content.DialogInterface.BUTTON_NEGATIVE;
+import static android.content.DialogInterface.BUTTON_POSITIVE;
+
+public class DeleteRepeatDialogFragment extends DialogFragment {
+
+    protected FormController formController = Collect.getInstance().getFormController();
```

Comment:
I'm not loving this protected field pattern for testing and I see it shows up in #3792 as well. I think that again, using a `ViewModel` here would help address that.

Reply:
I also feel now implementing a `ViewModel` would be better. I will do that.